### PR TITLE
CHANGE: enable performance test to be run continiously on U6

### PR DESCRIPTION
**Pull request by stefanunity**
### Description

This enables our current performance tests to be run continiously (daily on develop, weekly on master).

### Changes made

- created job that runs only the performance tests on all platforms (except tvOS)
  - these can also be run individually
- performance only test get triggered to run daily on develop and weekly on master branch
- updated com.unity.test-framework to 1.4.5 (from 1.4.3)
- updated com.unity.test-framework.performance to 3.0.3 (from 3.0.2)
- edits to shorten CI job names so they don't get clipped by yamato UI/better readility

### Testing

- run All Tests jobs successfully
- run all performance jobs tests successfully 
  - performance data does get submitted to observer
- NOTICE: there are some yellow yamato jobs. this happens because I renamed them and the github branch protection still has the old names. see All Tests job for all the green. once this lands I change the required jobs in git branch protection.

### Risk
Very little risk, all tests still run, no changes to the tests themselves.

### Checklist

Before review:

- [ ] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - JIRA ticket linked, example ([case %<ID>%](https://issuetracker.unity3d.com/product/unity/issues/guid/<ID>)). If it is a private issue, just add the case ID without a link.
    - Jira port for the next release set as "Resolved".
- [ ] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [ ] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

During merge:

- [ ] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

After merge:

- [ ] Create forward/backward port if needed. If you are blocked from creating a forward port now please add a task to ISX-1444.

### CODE REVIEW

File: .yamato/upm-ci.yml [hunk 1]
```diff
@@ -230,6 +238,31 @@ all_tests:
...
+    - .yamato/upm-ci.yml#run_ios_{{ editor.version }}_performance
+    {% endfor %}
+  triggers:
+    recurring:
+        - branch: develop
+          frequency: daily
+          rerun: on_new_revision            
+        - branch: master
+          frequency: weekly
+          rerun: on_new_revision
```
Comment (jfreire-unity) [inline .yamato/upm-ci.yml]: There's no `master` branch, only `stable` AFAIK.

**Review by jfreire-unity — CHANGES_REQUESTED** [review]
Change required since we don't have a `master` branch. 

I'm ok with approving once that's fixed. 

However, could just explain what's the goal with having this in CI? It's not clear to me.

More precisely, it's unclear to me what is the goal of running this performance jobs daily in case there are no changes to the package?  Is the goal to test against different Editor patch versions ?

**Comment by stefanunity** [comment]
> More precisely, it's unclear to me what is the goal of running this performance jobs daily in case there are no changes to the package? Is the goal to test against different Editor patch versions ?

These test are currently almost never run on develop. Since nobody runs test on develop, except for me before a release.
If we want to start making performance improvements and look at these tests for PRs too, then we need a baseline to compare against.
This baseline needs to be more then "one run from 2 months ago". Running these regularily on develop will hopefully give is that baseline.

File: .yamato/upm-ci.yml (hunk 1 above)
Comment (stefanunity) [reply .yamato/upm-ci.yml]: https://github.com/Unity-Technologies/InputSystem/pull/1996/commits/3f56cd946f5834b19048c9729c52b590abc34c44

**Review by jfreire-unity — APPROVED** [review]

**Review by lyndon-unity — APPROVED** [review]
Looks like a good update overall.

**Pull request merged**